islandora_openseadragon.module

```diff
@@ -76,148 +88,23 @@ function islandora_openseadragon_islandora_viewer_info() {
  * @param array $params
  *   Params required by the theme. We require the keys:
  *   - jp2_url: The URL to the JP2 image.
- * @param IslandoraAbstractObject $fedora_object
+ * @param AbstractObject $object
  *   The object that we are viewing.
  *
  * @return string
- *   HTML repersentation of the OpenSeadragon viewer
+ *   HTML representation of the OpenSeadragon viewer
  */
-function islandora_openseadragon_callback($params = NULL, $fedora_object = NULL) {
-  if (isset($params['jp2_url'])) {
-    $uri = $params['jp2_url'];
-    return theme('islandora_openseadragon_viewer', array('uri' => $uri, 'fedora_object' => $fedora_object));
-  }
-}
-
-/**
- * Implements template_preprocess_HOOK().
- */
-function template_preprocess_islandora_openseadragon_viewer(&$variables) {
-  // Variable fedora_object included in results, but not used.
-  $library_path = libraries_get_path('openseadragon');
-  $module_path = drupal_get_path('module', 'islandora_openseadragon');
-  $variables['viewer_id'] = 'islandora-openseadragon';
-  $settings = array(
-    'id' => $variables['viewer_id'],
-    'prefixUrl' => file_create_url("$library_path/images/"),
-  );
-
+function islandora_openseadragon_callback(array $params = array(), AbstractObject $object = NULL) {
   module_load_include('inc', 'islandora_openseadragon', 'includes/utilities');
-  $installed_version = islandora_openseadragon_get_installed_version();
-
-  if (version_compare($installed_version, '0.9.82') <= 0) {
-    // XXX: Version numbers don't quite line-up for the change in the nav image
-    // prefix URL, so now, let the hacking begin!... Just for compatibility
-    // with older versions.
-    $types = array('zoomIn', 'zoomOut', 'home', 'fullpage', 'previous', 'next');
-    $actions = array(
-      'REST' => 'rest',
-      'GROUP' => 'grouphover',
-      'HOVER' => 'hover',
-      'DOWN' => 'pressed',
-    );
-    foreach ($types as $type) {
-      foreach ($actions as $action => $namepart) {
-        $settings['navImages'][$type][$action] = format_string('!type_!action.png', array(
-          '!type' => strtolower($type),
-          '!action' => $namepart,
-        ));
-      }
-    }
+  $tile_sources = array();
+  if ($params) {
+    $matches = array();
+    preg_match('/datastream\/([^\/]*)\/view/', $params['jp2_url'], $matches);
```

Comment:
This is a comment, but why do we need this checking logic here? I'm a bit in the impression that the modules that implement JP2 will build $params['jp2_url'] and add the token and stuff like what happens in the [islandora_solution_pack_large_image ](https://github.com/Islandora/islandora_solution_pack_large_image/blob/7.x/islandora_large_image.module#L198-L223)
```PHP
if (isset($islandora_object['JP2']) && islandora_datastream_access(ISLANDORA_VIEW_OBJECTS, $islandora_object['JP2'])) {
    // Get token to allow access to XACML protected datastreams.
    // Always use token authentication in case there is a global policy.
    module_load_include('inc', 'islandora', 'includes/authtokens');
  $jp2_url = url("islandora/object/{$islandora_object->id}/datastream/JP2/view",
      array(
        'absolute' => TRUE,
        'query' => array('token' => $token),
      ));
    // Display large image.
    $params['jp2_url'] = $jp2_url;
  }

  $viewer = islandora_get_viewer($params, 'islandora_large_image_viewers', $islandora_object);
```

, so can we skip the regex and dsid check?